streaming/base/dataset.py

```diff
@@ -804,12 +804,27 @@ def load_state_dict(self, obj: Dict[str, Any]) -> None:
         Args:
             obj (Dict[str, Any]): The state.
         """
+        # Set shared memory block to be 1024 characters long. This enables calling
+        # `load_state_dict` multiple times without needing to resize the shared memory block.
+        # Resizing the shared memory block is not possible, and closing the shared memory block
+        # and replacing it with a new one is causing great difficulties.
+
         name = _get_path(self._shm_prefix_int, RESUME)
-        data = json.dumps(obj, sort_keys=True).encode('utf-8')
+        data = json.dumps(obj, sort_keys=True)
+
+        len_needed = len(data)
+        if len_needed > RESUME_SHM_SIZE:
+            raise ValueError(f'State dict for resumption is too large to fit in shared memory. ' +
+                             f'Please reduce the state dict size to less than {RESUME_SHM_SIZE} ' +
```

Comment:
More like "Please increase the state dict shared memory buffer size to be at least {len(data) + 1} bytes"

Reply:
changed this!